### PR TITLE
Support new protocol in Resolwe 26.x

### PR DESCRIPTION
**Pull request by jkokosar**
## Checklist

* [ ] Update CHANGELOG.rst for each commit separately:
  * Pay attention to write entries under the "Unreleased" section.
  * Mark all breaking changes as "**BACKWARD INCOMPATIBLE:**" and put them
    before non-breaking changes.
  * If a commit modifies a feature listed under "Unreleased" section,
    it might be sufficient to modify the existing CHANGELOG entry from previous
    commit(s).
* [ ] Bump the process version:
  * **MAJOR version (first number)**: Backward incompatible changes (changes
    that break the api/interface). Examples: renaming the input/output, adding
    mandatory input, removing input/output...
  * **MINOR version (middle number)**: add functionality or changes in a
    backwards-compatible manner. Examples: add output field, add non-mandatory
    input parameter, use a different tool that produces same results...
  * **PATCH version (last number)**: changes/bug fixes that do not affect
    the api/interface. Examples: typo fix, change/add warning messages...
* [ ] All inputs are used in process.
* [ ] All output fields have their ``re-save`` calls.

### Processes ###

* Code that was once accepted should be kept like it is. It should be changed
  only when its functionality has changed. Small parts may be changed
  to fix bugs (by changing code only to make it nicer you might introduce a bug).
* Set the optimal number of cores that the tool can use (check if parallelization
  can be performed).
* Description of tool should be self explanatory with links to tool's homepage
  and publication.
* Write explanatory input labels and add short command line parameter names
  in square brackets (for example: Maximum length [``--maxlength``]).
* Where possible provide default values for the process input fields.
  If they are provided, then ``required: false`` is not needed.
* Add descriptive error and warning messages.
* Place a ``re-checkrc`` call after each process step (call of a tool, a
  Bash command, a Python script or an R script).
* Add ``re-progress`` calls where appropriate.

### Commit and PR messages ###

* Each commit should be minimal (i.e. 1 change) and self-contained (including
  tests).
* Changes to Docker images need to be accepted in a separate pull request.
  Before the new images can be used for automated testing of your PR, they
  must be manually tagged and pushed to Docker Hub.
* Mark incomplete PRs with the [WIP] (Work In Progress) tag.
* Do not paste links to private repositories from the public one.

### Tests ###

* Add meaningful names of test files with spaces:
  * Genomes: ``genome species.fa.gz``
  * Annotations: ``annotation species.gtf.gz``
  * Adapters: ``adapters-name/type-source``.
* Output and specific input files should include the name of the tested tool.
* Keep test files' size small. If there is no other option, add the large
  test file to the ``files/large`` directory so it will be tracked with Git LFS.
* Pay attention to edge cases: what will happen if any data is
  missing, if length of some list is 0, if string is given where one expects
  integer, if file is empty, ...
* When testing workflows, try to verify workflow success by checking one
  of the last steps. Pick the one that is most likely to fail. Also test
  that all steps have 'OK' status.

### CODE REVIEW

**Review by gregorjerse — APPROVED** [review]

File: resolwe_bio/processes/import_data/seq_reads.yml [hunk 1]
```diff
@@ -105,6 +105,7 @@
         FASTQC_URL+=("{\"file\":\"fastqc/${NAME}_fastqc/fastqc_report.html\",\"refs\":[\"fastqc/${NAME}_fastqc\"]}")
       done
 
+      re-save-file-list fastq "${FASTQGZ[@]}"
```
Comment (romunov) [inline resolwe_bio/processes/import_data/seq_reads.yml:108]: Is the one in line 66 not enough?

File: resolwe_bio/processes/variant_calling/gatk.yml [hunk 2]
```diff
@@ -77,12 +81,15 @@
     runtime: polyglot
     language: bash
     program: |
+      pip3 install --user resolwe-runtime-utils
```
Comment (romunov) [inline resolwe_bio/processes/variant_calling/gatk.yml:84]: Couldn't we fork the docker image and add this there? This feels very hackish...

File: resolwe_bio/tests/processes/test_upload.py [hunk 3]
```diff
@@ -180,7 +180,8 @@ def test_upload_paired_end_reads(self):
         wrong_mates = self.run_process("upload-fastq-paired", inputs, Data.STATUS_ERROR)
         error_msg = [
             "The number of mate-pair files in split-lane samples must match. 2 and 1 "
-            "input files were given for the -fq and -fq2 inputs, respectively."
+            "input files were given for the -fq and -fq2 inputs, respectively.",
```
Comment (romunov) [inline resolwe_bio/tests/processes/test_upload.py:183]: I think there should be a space before string closes.

File: resolwe_bio/tests/processes/test_upload.py [hunk 4]
```diff
@@ -192,7 +193,8 @@ def test_upload_paired_end_reads(self):
             "upload-fastq-paired", inputs, Data.STATUS_ERROR
         )
         error_msg = [
-            "Non-unique input file names detected: ['mate1_reordered.fastq.gz']."
+            "Non-unique input file names detected: ['mate1_reordered.fastq.gz'].",
```
Comment (romunov) [inline resolwe_bio/tests/processes/test_upload.py:196]: I think there should be a space before string closes.

File: resolwe_bio/tests/processes/test_upload.py [hunk 5]
```diff
@@ -206,7 +208,8 @@ def test_upload_paired_end_reads(self):
         error_msg = [
             "Format error in mate-pairs mate1_diff_num_reads.fastq.gz and mate2_diff_num_reads.fastq.gz. "
             "Error in sequence file at unknown line: Reads are improperly paired. "
-            "There are more reads in file 2 than in file 1."
+            "There are more reads in file 2 than in file 1.",
```
Comment (romunov) [inline resolwe_bio/tests/processes/test_upload.py:211]: I think there should be a space before string closes.

File: resolwe_bio/tests/processes/test_upload.py [hunk 6]
```diff
@@ -220,7 +223,8 @@ def test_upload_paired_end_reads(self):
         error_msg = [
             "Format error in mate-pairs mate1_reordered.fastq.gz and mate2_reordered.fastq.gz. "
             "Error in sequence file at unknown line: Reads are improperly paired. Read "
-            "name 'read1/1 some text' in file 1 does not match 'read2/2' in file 2."
+            "name 'read1/1 some text' in file 1 does not match 'read2/2' in file 2.",
```
Comment (romunov) [inline resolwe_bio/tests/processes/test_upload.py:226]: I think there should be a space before string closes.

File: resolwe_bio/tests/processes/test_upload.py [hunk 7]
```diff
@@ -282,15 +286,19 @@ def test_upload_single_end_reads(self):
         empty_input = self.run_process(
             "upload-fastq-single", {"src": ["empty.fastq.gz"]}, Data.STATUS_ERROR
         )
-        error_msg = ["Input file empty.fastq.gz contains no read sequences."]
+        error_msg = [
+            "Input file empty.fastq.gz contains no read sequences.",
```
Comment (romunov) [inline resolwe_bio/tests/processes/test_upload.py:290]: I think there should be a space before string closes.

File: resolwe_bio/tests/processes/test_upload.py [hunk 8]
```diff
@@ -282,15 +286,19 @@ def test_upload_single_end_reads(self):
...
+        ]
         self.assertEqual(empty_input.process_error, error_msg)
 
         garbage_input = self.run_process(
             "upload-fastq-single", {"src": ["garbage.fastq.gz"]}, Data.STATUS_ERROR
         )
         error_msg = [
             "Error in file garbage.fastq.gz. Error in FASTQ file at line 1: Line expected "
-            "to start with '@', but found 'S'"
+            "to start with '@', but found 'S'",
```
Comment (romunov) [inline resolwe_bio/tests/processes/test_upload.py:300]: I think there should be a space before string closes or a new line feed.

File: resolwe_bio/tests/processes/test_upload.py [hunk 9]
```diff
@@ -299,8 +307,9 @@ def test_upload_single_end_reads(self):
         )
         error_msg = [
             "Error in file garbage2.fastq.gz. Error in FASTQ file at line 3: Sequence descriptions "
-            "don't match ('Some random content' != '+++'). The second sequence description must "
-            "be either empty or equal to the first description."
+            "don't match ('Some random content' != '+++').\nThe second sequence description must "
+            "be either empty or equal to the first description.",
```
Comment (romunov) [inline resolwe_bio/tests/processes/test_upload.py:311]: I think there should be a space before string closes.

**Review by romunov — APPROVED** [review]

File: resolwe_bio/processes/import_data/seq_reads.yml (hunk 1 above)
Comment (jkokosar) [reply resolwe_bio/processes/import_data/seq_reads.yml:108]: The code transcodes the reads quality information if the old encoding is detected. For technical reasons, the output field needs to be referenced again in such cases.

File: resolwe_bio/tests/processes/test_upload.py (hunk 3 above)
Comment (jkokosar) [reply resolwe_bio/tests/processes/test_upload.py:183]: There are two elements in the error msg list object. It's not one string.

**Pull request merged**